src/foam/core/XML.js

```diff
@@ -0,0 +1,466 @@
+      } else {
+        this.output(o);
+      }
+      var ret = this.buf_;
+      this.reset(); // reset to avoid retaining garbage
+      return ret;
+    },
+
+    {
+      name: 'createFObj',
+      code: foam.mmethod({
+        Object: function (o) {
+          // Create FObject
+          var className = o.className;
+          var obj = foam.lookup(className).create();
+          var props = o.children;
+
+          // Populate FObject with properties
+          for ( var propIndex = 0; propIndex < props.length; propIndex++ ) {
+
+            var currentProp = props[propIndex];
+            var prop = obj.cls_.getAxiomByName(currentProp.tagName);
+            var childName = currentProp.firstChild.localName;
+            // Specific case for array
+            if ( currentProp.className === 'Array' ) {
+              // Array of FObjects
+              if ( childName === 'object' ) {
+                var nestObjArray = Array.from(currentProp.childNodes);
+                prop.set(obj, this.createFObj(nestObjArray));
+              } else {
+                // Array of other objects with 'value' tag names
+                var arrayValue = (Array.from(currentProp.children)).map( function (x) { return x.innerHTML; });
+                prop.set(obj, arrayValue);
+              }
+              continue;
+            }
+
+            // Nested Object
+            if ( childName === 'object' ) {
+              var nestObj = this.createFObj(currentProp.firstChild);
+              prop.set(obj, nestObj);
+              continue;
+            }
+
+            // Regular Prop
+            if ( currentProp.firstChild.nodeValue ) {
+              var val = currentProp.firstChild.nodeValue.replace(/\"/g, "");
+              prop.set(obj, prop.of ? foam.lookup(prop.of.id).create({ ordinal: val }) : val );
+            } else if ( currentProp.firstChild.innerHTML ) {
+              var v = currentProp.firstChild.innerHTML.replace(/\"/g, "");
+              prop.set(obj, prop.of ? foam.lookup(prop.of.id).create({ ordinal: v }) : v );
+            }
+          }
+          return obj;
+        },
+        Array: function (o) {
+          var fObjects = []
+          for ( index = 0; index < o.length; index++ ) {
+              fObjects.push(this.createFObj(o[index]));
+          }
+          return fObjects;
+        }
+      })
+    },
+
+    {
+      name: 'objectify',
+      code: foam.mmethod({
+        String: function (o) {
+          if ( o ) {
+            // Convert xml string into an xml DOM object for node traversal
+            var parser = new DOMParser();
+            o = o.replace(/\t|\n|\r|↵/g, "");
+            var xmlDoc = parser.parseFromString(o, "text/xml");
+            var rootName = xmlDoc.firstChild.nodeName;
+            // Check if multiple objects
+            if ( rootName === 'objects' ) {
+              return this.createFObj(Array.from(xmlDoc.firstChild.childNodes));
+            } else if ( rootName === 'object' ) {
```

Comment:
On line 358, the check is for FObjects that are nested within the object itself. If it's not an array of nested FObjects, then it will be primitive typed values that have to be parsed. In the case of line 411, it is checking whether the XML string contains multiple objects or a single object to parse.